frontend/src/Components/HomePage/UpcomingTrips/SeatBidding.jsx

```diff
@@ -0,0 +1,231 @@
+import { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import axios from 'axios';
+import { io } from 'socket.io-client';
+import { API_BASE_URL } from '../../../config';
+import './SeatBidding.css';
+
+/**
+ * SeatBidding component for managing seat bidding functionality
+ * @param {Object} props - Component props
+ * @param {Object} props.trip - Trip information
+ * @param {string} props.flightId - ID of the flight
+ * @param {Function} props.onBiddingCompleted - Callback function when bidding is completed
+ */
+const SeatBidding = ({ trip, flightId, onBiddingCompleted }) => {
+    const [activeBiddings, setActiveBiddings] = useState([]);
+    const [bidAmounts, setBidAmounts] = useState({});
+    const [isLoading, setIsLoading] = useState(true);
+    const [message, setMessage] = useState('');
+
+    /**
+     * Fetch active biddings for the current flight
+     */
+    const fetchActiveBiddings = async () => {
+        try {
+            const response = await axios.get(`${API_BASE_URL}/bidding/active/${flightId}`, { withCredentials: true });
+            setActiveBiddings(response.data);
+        } catch (error) {
+            // Error handling for failed fetch
+        } finally {
+            setIsLoading(false);
+        }
+    };
+
+    useEffect(() => {
+        const newSocket = io(API_BASE_URL, { withCredentials: true });
+
+        newSocket.on('connect', () => newSocket.emit('join-flight', flightId));
+        newSocket.on('new-bid', (data) => {
+            fetchActiveBiddings();
+            handleNewBid(data);
+        });
+        newSocket.on('bidding-completed', handleBiddingCompleted);
+        newSocket.on('bidding-expired', handleBiddingExpired);
+        newSocket.on('bidding-cancelled', handleBiddingCancelled);
+
+        fetchActiveBiddings();
+
+        // Cleanup function
+        return () => {
+            newSocket.emit('leave-flight', flightId);
+            newSocket.disconnect();
+        };
+    }, [flightId]);
+
+    /**
+     * Handle new bid event
+     * @param {Object} data - New bid data
+     */
+    const handleNewBid = ({ biddingId, bid }) => {
+        setActiveBiddings(prev => prev.map(bidding =>
```

Comment:
instead of mapping over the full list, I wonder if keeping a data structure that allows direct access to placed bids based on biddingId

Reply:
Thanks for pointing it out. I'm focusing on making the program work for now. Will definitely look into optimizing it later, hopefully next week. What data structure do you think I could use here?

Reply:
I would use a map so you don't have to go through the full array